ownCloudAppShared/Client/User Interface/SortBar.swift

```diff
@@ -93,57 +139,82 @@ public class SortBar: UIView, Themeable, UIPopoverPresentationControllerDelegate
 			}
 			updateSortButtonTitle()
 
-			sortButton?.accessibilityLabel = NSString(format: "Sort by %@".localized as NSString, sortMethod.localizedName()) as String
+			sortButton?.accessibilityLabel = NSString(format: "Sort by %@".localized as NSString, sortMethod.localizedName) as String
 			sortButton?.sizeToFit()
 
 			if let oldSementIndex = SortMethod.all.index(of: oldValue) {
-				sortSegmentedControl?.setTitle(oldValue.localizedName(), forSegmentAt: oldSementIndex)
+				sortSegmentedControl?.setTitle(oldValue.localizedName, forSegmentAt: oldSementIndex)
 			}
 			if let segmentIndex = SortMethod.all.index(of: sortMethod) {
 				sortSegmentedControl?.selectedSegmentIndex = segmentIndex
-				sortSegmentedControl?.setTitle(sortDirectionTitle(sortMethod.localizedName()), forSegmentAt: segmentIndex)
+				sortSegmentedControl?.setTitle(sortDirectionTitle(sortMethod.localizedName), forSegmentAt: segmentIndex)
 			}
 
 			delegate?.sortBar(self, didUpdateSortMethod: sortMethod)
 		}
 	}
 
+	public var searchScope : SearchScope {
+		didSet {
+			delegate?.searchScope = searchScope
+			searchScopeSegmentedControl?.selectedSegmentIndex = searchScope.rawValue
+		}
+	}
+
 	// MARK: - Init & Deinit
 
-	public init(frame: CGRect, sortMethod: SortMethod) {
+	public init(frame: CGRect, sortMethod: SortMethod, searchScope: SearchScope = .local) {
 		sortSegmentedControl = SegmentedControl()
 		selectButton = UIButton()
 		sortButton = UIButton(type: .system)
+		searchScopeSegmentedControl = SegmentedControl()
 
 		self.sortMethod = sortMethod
+		self.searchScope = searchScope
 
 		super.init(frame: frame)
 
-		if let sortButton = sortButton, let sortSegmentedControl = sortSegmentedControl, let selectButton = selectButton {
+		if let sortButton = sortButton, let sortSegmentedControl = sortSegmentedControl, let searchScopeSegmentedControl = searchScopeSegmentedControl, let selectButton = selectButton {
 			sortButton.translatesAutoresizingMaskIntoConstraints = false
 			sortSegmentedControl.translatesAutoresizingMaskIntoConstraints = false
 			selectButton.translatesAutoresizingMaskIntoConstraints = false
+			searchScopeSegmentedControl.translatesAutoresizingMaskIntoConstraints = false
 
 			sortButton.accessibilityIdentifier = "sort-bar.sortButton"
 			sortSegmentedControl.accessibilityIdentifier = "sort-bar.segmentedControl"
+			searchScopeSegmentedControl.accessibilityIdentifier = "sort-bar.searchScopeSegmentedControl"
+
+			for scope in SearchScope.allCases {
+				if let image = scope.image {
+					searchScopeSegmentedControl.insertSegment(with: image, at: scope.rawValue, animated: false)
```

Comment:
Does it make more sense to only show the titles? Your chosen images are great, but it looks a little bit to large for the small segmented control UI element. But needs to check, if the width fits for all UI elements in this row on a small iPhone display.
Can you try a build with titles only please?

Reply:
I did initially try with titles, but ran into space problems and had issues finding a short word that explains the purpose as well as the icons. "all" fits, but "everywhere" or "all folders" would be better. Same with "folder" vs. "this folder". But then the UI no longer fits.

If you feel the icons are too large, we could also try slightly scaled down versions. What do you think?

Reply:
I just tested the label solution on a small screen size (iPhone 6) and it fits. I also discovered, that the `Multiselect` button needs to be hidden, when search is active, because the UI elements are hidden in this mode. This also give us more space for the search scope segmented control.
Maybe it makes sense to use the label `Account` instead of `All` for the global search scope.

If you agree, I can push my changes, but I'm also interested in your thoughts.
Here you can see the result:

![Simulator Screen Shot - iPhone 6s - 2021-04-09 at 11 12 55](https://user-images.githubusercontent.com/736109/114158139-a91c2b00-9924-11eb-8178-cf7e562b8057.png)

Reply:
"Account" and "Folder" are good choices for labels IMO.

My worry, however, is with iOS 12 on the smallest supported devices (iPhone 5s). 

Even in English, it's already getting really tight, but in German there are overlaps:

English | German
--------|--------
![Simulator Screen Shot - iPhone 5s - 2021-04-09 at 14 58 17](https://user-images.githubusercontent.com/639669/114183479-05db0e00-9944-11eb-9f73-5cbd46ac0a61.png) | ![Simulator Screen Shot - iPhone 5s - 2021-04-09 at 15 04 08](https://user-images.githubusercontent.com/639669/114184194-d7116780-9944-11eb-8019-d95b3b54c7eb.png)

Reply:
I also just noticed that we don't change the placeholder text when switching between account and folder scope (always "Search this folder"). Will fix that.

Reply:
On the other hand, we can't use the SF Symbols on iOS 12, so would either need to provide our own icons - or drop back to text there. Taking into account that iOS 12 + iPhone 5s should be a really small cohort, the only issue for these is purely cosmetic in nature, and that text communicates the purpose a lot clearer, I came to the conclusion that we should indeed use text here and merged the changes from your branch.

Also fixed the placeholder text.